pkg/sideinputs/store/interface.go

```diff
@@ -0,0 +1,21 @@
+package store
+
+import (
+	"context"
+)
+
+// SideInputKVEntry defines what can be read on the Watch stream.
+type SideInputKVEntry interface {
+	// Key is the key that was retrieved.
+	Key() string
+	// Value is the retrieved value.
+	Value() []byte
+}
+
+// SideInputWatcher watches the KV bucket for SideInput data updates.
+type SideInputWatcher interface {
+	// Watch starts the watermark kv watcher and returns a kv updates channel and a watcher stopped channel.
+	Watch(context.Context) (<-chan SideInputKVEntry, <-chan struct{})
+	GetKVName() string
+	Close()
+}
```

Comment:
let's use a generic interface KVEntry and KVWatcher. So both watermark and side input can implement those. WDYT?